packages/std/src/init_handle.rs

```diff
@@ -190,11 +224,69 @@ where
     }
 }
 
+impl<T> From<Context<T>> for MigrateResponse<T>
+where
+    T: Clone + fmt::Debug + PartialEq + JsonSchema,
+{
+    fn from(ctx: Context<T>) -> Self {
+        MigrateResponse {
+            messages: ctx.messages,
+            log: ctx.log,
+            data: ctx.data,
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Context<T = Empty>
+where
+    T: Clone + fmt::Debug + PartialEq + JsonSchema,
+{
+    messages: Vec<CosmosMsg<T>>,
+    log: Vec<LogAttribute>,
+    data: Option<Binary>,
+}
+
+impl<T> Default for Context<T>
+where
+    T: Clone + fmt::Debug + PartialEq + JsonSchema,
+{
+    fn default() -> Self {
+        Context {
+            messages: vec![],
+            log: vec![],
+            data: None,
+        }
+    }
+}
+
+impl<T> Context<T>
+where
+    T: Clone + fmt::Debug + PartialEq + JsonSchema,
+{
+    pub fn new() -> Self {
+        Context::default()
+    }
+
+    pub fn add_event<K: ToString, V: ToString>(&mut self, key: K, value: V) {
+        self.log.push(log(key, value));
```

Comment:
Is "event" the term we want to use instead of "log"? Then we should do that in the whole codebase. No need to do it here but let's agree on that the desired term is.

Reply:
Good question.

I will stick with log now and be consistent, and maybe we change this everywhere before 1.0